hledger/Hledger/Cli/Commands/Aregister.hs

```diff
@@ -200,18 +210,13 @@ accountTransactionsReportItemAsText
     -- gather content
     accts = -- T.unpack $ elideAccountName acctwidth $ T.pack
             otheracctsstr
-    amt = TL.toStrict . TB.toLazyText . wbBuilder $ showamt amtwidth change
-    bal = TL.toStrict . TB.toLazyText . wbBuilder $ showamt balwidth balance
-    showamt w = showMixedAmountB noPrice{displayColour=color_, displayMinWidth=Just w, displayMaxWidth=Just w}
-    -- alternate behaviour, show null amounts as 0 instead of blank
-    -- amt = if null amt' then "0" else amt'
-    -- bal = if null bal' then "0" else bal'
-    (amtlines, ballines) = (T.lines amt, T.lines bal)
-    (amtlen, ballen) = (length amtlines, length ballines)
-    numlines = max 1 (max amtlen ballen)
-    (amtfirstline:amtrest) = take numlines $ amtlines ++ repeat "" -- posting amount is top-aligned
-    (balfirstline:balrest) = take numlines $ replicate (numlines - ballen) "" ++ ballines -- balance amount is bottom-aligned
-    spacer = T.replicate (totalwidth - (amtwidth + 2 + balwidth)) " "
+    amt = showamt change
+    bal = showamt balance
+    showamt = showMixedAmountLinesB noPrice{displayColour=color_}
+    -- Since this will usually be called with the knot tied between this(amt|bal)width and
+    -- preferred(amt|bal)width, make sure the former do not depend on the latter to avoid loops.
+    thisamtwidth = maximumDef 0 $ map wbWidth amt
+    thisbalwidth = maximumDef 0 $ map wbWidth bal
```

Comment:
Maybe this should be commented out too, like the knot-tying above ? I see it's being used though.

Reply:
I think I've concluded that the knot-tying can't be done in this way without causing a huge space leak. That probably means I can remove all that commented-out code, but maybe this should be done along with the register code as a separate PR.

Reply:
I didn't quite understand. For this PR, shouldn't we at least change this comment, since knot tying is currently not being used ?

Reply:
I've removed the relevant bits from Aregister.hs, and added another commit which removes the equivalent code in Register.hs.